inst/shiny/modules/tab_nca/setup/settings.R

```diff
@@ -0,0 +1,489 @@
+                  ns("bioavailability"),
+                  "Calculate Bioavailability:",
+                  choices = pknca_cdisc_terms %>%
+                    filter(startsWith(PPTESTCD, "FABS_") | startsWith(PPTESTCD, "FREL_")) %>%
+                    pull(PKNCA, PPTESTCD),
+                  multiple = TRUE,
+                  selected = NULL
+                )
+              )
+            )
+          )
+        )
+      ),
+      accordion_panel(
+        title = "Parameter Selection",
+        reactableOutput(ns("nca_parameters")),
+        card(
+          full_screen = FALSE,
+          style = "margin-top: 2em;",
+          card_header("Selected NCA Parameters"),
+          card_body(uiOutput(ns("nca_param_display"))
+          )
+        )
+      ),
+      accordion_panel(
+        title = "Data Imputation",
+        input_switch(
+          id = ns("should_impute_c0"),
+          label = "Impute Concentration",
+          value = TRUE
+        ),
+        br(),
+        helpText(HTML(paste(
+          "Imputes a start-of-interval concentration to calculate non-observational parameters:",
+          "- If DOSNO = 1 & IV bolus: C0 = 0",
+          "- If DOSNO > 1 & not IV bolus: C0 = predose",
+          "- If IV bolus & monoexponential data: logslope",
+          "- If IV bolus & not monoexponential data: C0 = C1",
+          sep = "<br>"
+        )))
+      ),
+      accordion_panel(
+        title = "Partial AUCs",
+        reactableOutput(ns("auc_table")),
+        actionButton(ns("addRow"), "Add Row")
+      ),
+      accordion_panel(
+        title = "Flag Rule Sets",
+        .rule_input(ns("adj.r.squared"), "RSQADJ:", 0.7, 0.05, 0, 1),
+        .rule_input(ns("aucpext.obs"), "AUCPEO (% ext.observed):", 20, 1, 0, 100),
+        .rule_input(ns("aucpext.pred"), "AUCPEP (% ext.predicted):", 20, 5, 0, 100),
+        .rule_input(ns("span.ratio"), "SPAN:", 2, 1, 0)
+      ),
+      id = "acc",
+      open = c("General Settings", "Parameter Selection")
+    )
+  )
+}
+
+settings_server <- function(id, data, adnca_data, settings_override) {
+  moduleServer(id, function(input, output, session) {
+    ns <- session$ns
+
+    conc_data <- reactive(adnca_data()$conc$data)
+
+    # File Upload Handling
+    observeEvent(settings_override(), {
+      settings <- settings_override()
+
+      log_debug_list("User settings override:", settings)
+
+      not_compatible <- c()
+
+      # General #
+      if (all(settings$analyte %in% unique(data()$PARAM))) {
+        updateSelectInput(inputId = "select_analyte", selected = settings$analyte)
+      } else {
+        not_compatible <- append(not_compatible, "Analyte")
+      }
+
+      if (all(settings$doseno %in% unique(data()$DOSNO))) {
+        updateSelectInput(inputId = "select_doseno", selected = settings$doseno)
+      } else {
+        not_compatible <- append(not_compatible, "Dose Number")
+      }
+
+      if (all(settings$pscpec %in% unique(data()$PCSPEC))) {
```

Comment:
is there a typo here in `pscpec` or `PCSPEC` ? they are written differently from each other.